common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java

```diff
@@ -333,33 +333,63 @@ protected Path getRecoveryPath(String fileName) {
   }
 
   /**
+   * Check the chosen DB file available or not.
+   */
+  protected Boolean checkFileAvailable(File file) {
```

Comment:
I'm not sure if it is a thorough way to check disk healthy, in our internal case, we found that disk is not mounted (due to failure), and trying to write to this unmounted disk throws permission deny exception.

I'm thinking that disk unwritable is just one case of disk unhealthy, maybe we should check YARN's disk healthy check mechanism.